sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/ConditionEvaluatesToConstant.CSharp8.cs

```diff
@@ -0,0 +1,129 @@
+﻿using System;
+
+namespace Tests.Diagnostics
+{
+    public interface IWithDefaultImplementation
+    {
+        void DoSomething();
+
+        //Default interface methods
+        void DoNothing()
+        {
+            var a = false;
+            if (a)  // Noncompliant
+            {       // Secondary
+                DoSomething(); // never executed
+            }
+        }
+    }
+
+    public class CSharp8
+    {
+        int SwitchExpression()
+        {
+            var a = false;
+            return a switch { true => 0, false => 1 };  // FN - switch arms are not inspected
+        }
+
+        int SwitchExpression_Discard()
+        {
+            var a = false;
+            return a switch { true => 0, _ => 1 };  // FN - switch arms are not inspected
+        }
+
+        int SwitchExpression_Results()
+        {
+            bool a;
+
+            a = false switch { _ => false };
+            if (a)  // Noncompliant
+            {       // Secondary
+                return 42; // never executed
+            }
+
+            a = false switch { false => false, _ => false };
+            if (a)  // Noncompliant
+            {       // Secondary
+                return 42; // never executed
+            }
+
+            a = false switch { true => true, _ => false };
+            if (a)  // FN - switch arms are not constrained
```

Comment:
In the example above the arms are considered and here not? 😄 

I think this case is different from the others since we miss to add a constraint for the governing expression (not switch arms like it's mentioned)

Reply:
In the example above, the arms produce the same value. So the result of the test does not depend on constraints (not now, nor in the future).

I would keep the comment like this. We don't need to document too much details in here. And it will get deleted when the CFG is fixed anyway.

Reply:
Maybe I miss something but shouldn't the false negative be on the switch expression line and not here?

Reply:
You're right. It should be on both.